sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -124,6 +124,52 @@ struct snd_soc_ops max_98373_ops = {
 };
 EXPORT_SYMBOL_NS(max_98373_ops, SND_SOC_INTEL_SOF_MAXIM_COMMON);
 
+/*
+ * Find present Maxim codecs on board
+ *
+ * amp_nums: return the codec numbers, present 2 or 4 codecs support.
+ */
+
+int get_maxim_num_codec(void)
+{
+	const char * const CODEC_NUMS[] = { "0", "1", "2", "3" };
+	const char * const MAXIM_DEVICES[] = {"MX98373", "MX98390"};
+
+	unsigned int codec_nums = 0;
+	struct acpi_device *adev;
+
+	/* finding the first matched codec device */
+	for (int i = 0; i < ARRAY_SIZE(MAXIM_DEVICES); i++) {
+		adev = acpi_dev_get_first_match_dev(MAXIM_DEVICES[i], NULL, -1);
+
+		/* not found: next device id. */
+		if (!adev)
+			continue;
+
+		/* found: retrieve how many present codec of matched device. */
+		while (codec_nums < ARRAY_SIZE(CODEC_NUMS)) {
+			adev = acpi_dev_get_first_match_dev(MAXIM_DEVICES[i],
+							    CODEC_NUMS[codec_nums], -1);
+			if (!adev)
+				goto checked_num_support;
```

Comment:
CODEC_NUMS is actually CODEC_UID, right? This doesn't work if you have discontinuous UIDs, like uid = 0, 2.
You can use 
```
for (j = 0; J < ARRAY_SIZE(CODEC_NUMS); j++) {
        ...
        if (!adev)
                continue;
        ... 
}
```